kotlin-ktor/src/test/java/com/baeldung/graphql/client/GraphQLClientTest.kt

```diff
@@ -0,0 +1,142 @@
+package com.baeldung.graphql.client
+
+import com.baeldung.graphql.server.data.AttendeeRepository
+import com.baeldung.graphql.server.data.ConferenceRepository
+import com.baeldung.graphql.server.graphQLModule
+import io.ktor.server.engine.*
+import io.ktor.server.netty.*
+import kotlinx.coroutines.runBlocking
+import org.junit.AfterClass
+import org.junit.BeforeClass
+import kotlin.test.*
+
+class GraphQLClientTest {
+
+    private val graphQLService = GraphQLService("http://0.0.0.0:8080/graphql")
+
+    // https://stackoverflow.com/questions/61245981/how-can-i-start-ktor-server-programmatically-for-integration-tests
+    // Since regular Ktor tests work with a dummy application with no server, we can't actually test the client properly without starting it
+    // So we start it up and tear it down for every test.
```

Comment:
Do you think this is something particularly confusing (and so worth mentioning) or something that Ktor users would already be familiar with as a test design pattern?

Reply:
The usual behavior for these tests is to use a dummy server, and it's a bit obscure to find data on setting up an application from scratch for tests